src/web/js/google-apis/api-wrapper.js

```diff
@@ -0,0 +1,358 @@
+        throw new APIResponseError(result);
+      }
+      return result;
+    });
+  }
+
+  /**
+   * Wraps the given Google query in a promise
+   * @param request - The request object to wrap (made by methods from `gapi')
+   * @param {boolean} [skipAuth] - If true, performs the request without
+   *        authentication.
+   * @returns {Promise} A promise which resolves to the result of the Google query
+   */
+  function gQ(request, skipAuth) {
+    var oldAccess = gapi.auth.getToken();
+    if (skipAuth) { gapi.auth.setToken({ access_token: null }); }
+    var ret = failCheck(authCheck(function() {
+      var d = Q.defer();
+      // TODO: This should be migrated to a promise
+      request.execute(function(result) {
+        d.resolve(result);
+      });
+      return d.promise;
+    }));
+    if (skipAuth) {
+      ret.fin(function() {
+        gapi.auth.setToken({ access_token: oldAccess });
+      });
+    }
+    return ret;
+  }
+
+  
+
+  /**
+   * Loads the Google API denoted by the given parameters
+   * @param {object} params - The designator for the API. Valid fields
+            are `name`, `url`, `version`, and `callback`
+   * @returns Nothing if `params.callback` is provided, otherwise a promise
+   *          which resolves to the loaded API/APIs.
+   */
+  function loadAPI(params) {
+    if (!params) {
+      throw new GoogleAPIError("Missing API loading parameters");
+    }
+    if (params.reauth) {
+      if (params.reauth.immediate === undefined) {
+        throw new GoogleAPIError("Missing required field to load(): "
+                                 + "params.reauth.immediate");
+      }
+      var reloaded = Q.defer();
+      loadAPIWrapper(params.reauth.immediate)
+        .then(function(gw) {
+          // Shallow copy
+          var copy = $.extend({}, params);
+          delete copy.reauth;
+          gw.load(copy)
+            .then(function(loaded) {
+              reloaded.resolve(loaded);
+            });
+        });
+      return reloaded.promise;
+    }
+    var preKeys = Object.keys(gapi.client);
+
+    function processKey(key) {
+      function processObject(obj, dest) {
+        Object.keys(obj).forEach(function(key) {
+          if (typeof(obj[key]) === "object") {
+            dest[key] = {};
+            processObject(obj[key], dest[key]);
+          } else if (typeof(obj[key]) === "function") {
+            // Use one argument, since it seems that's all that's
+            // used, and there's no `Function.prototype.apply` equivalent
+            // that doesn't change the function's context
+            var original = obj[key];
+            dest[key] = (function(args, skipAuth) {
+              return gQ(original(args), skipAuth);
+            });
+          } else {
+            dest[key] = obj[key];
+          }
+        });
+      }
+
+      if (Object.keys(_GWRAP_APIS).includes(key)) {
+        // Do nothing
+        return null;
```

Comment:
Oh, I think I missed the distinction between `processKey` and `processObject` (because the latter function is nested in the former, I kinda missed it).